pyface/ui/qt4/font.py

```diff
@@ -0,0 +1,202 @@
+    'swiss': QFont.SansSerif,
+    'monospace': QFont.Monospace,
+    'modern': QFont.Monospace,
+    'typewriter': QFont.TypeWriter,
+}
+
+weight_to_qt_weight = {
+    100: 0,
+    200: QFont.ExtraLight,
+    300: QFont.Light,
+    400: QFont.Normal,
+    500: QFont.Medium,
+    600: QFont.DemiBold,
+    700: QFont.Bold,
+    800: QFont.ExtraBold,
+    900: QFont.Black,
+    1000: 99,
+}
+qt_weight_to_weight = {
+    0: 'thin',
+    QFont.ExtraLight: 'extra-light',
+    QFont.Light: 'light',
+    QFont.Normal: 'normal',
+    QFont.Medium: 'medium',
+    QFont.DemiBold: 'demibold',
+    QFont.Bold: 'bold',
+    QFont.ExtraBold: 'extra-bold',
+    QFont.Black: 'black',
+    99: 'extra-heavy',
+}
+
+style_to_qt_style = {
+    'normal': QFont.StyleNormal,
+    'oblique': QFont.StyleOblique,
+    'italic': QFont.StyleItalic,
+}
+qt_style_to_style = {value: key for key, value in style_to_qt_style.items()}
+
+
+def font_to_toolkit_font(font):
+    """ Convert a Pyface font to a Qfont.
+
+    Parameters
+    ----------
+    font : pyface.font.Font
+        The Pyface font to convert.
+
+    Returns
+    -------
+    qt_font : QFont
+        The best matching Qt font.
+    """
+    qt_font = QFont()
+    families = []
+    default_family = None
+
+    for family in font.family:
+        if family not in generic_family_to_qt_family:
+            families.append(family)
+        elif default_family is None:
+            default_family = family
+
+    if families and hasattr(qt_font, 'setFamilies'):
+        # Qt 5.13 and later
+        qt_font.setFamilies(families)
+    elif families:
+        qt_font.setFamily(families[0])
+        # Note: possibily could use substitutions here,
+        # but not sure if global (which would be bad, so we don't)
+
+    if default_family is not None:
+        qt_font.setStyleHint(generic_family_to_qt_family[default_family])
+
+    qt_font.setPointSizeF(font.size)
+    qt_font.setWeight(weight_to_qt_weight[font.weight_])
+    qt_font.setStretch(font.stretch)
+    qt_font.setStyle(style_to_qt_style[font.style])
+    qt_font.setUnderline('underline' in font.variants)
+    qt_font.setStrikeOut('strikethrough' in font.variants)
+    qt_font.setOverline('overline' in font.variants)
+    if 'small-caps' in font.variants:
+        qt_font.setCapitalization(QFont.SmallCaps)
+    return qt_font
+
+
+def toolkit_font_to_properties(toolkit_font):
+    """ Convert a QFont to a dictionary of font properties.
+
+    Parameters
+    ----------
+    toolkit_font : QFont
+        The Qt QFont to convert.
+
+    Returns
+    -------
+    properties : dict
+        Font properties suitable for use in creating a Pyface Font.
+    """
+    family = []
+
+    if hasattr(toolkit_font, 'families'):
+        # Qt 5.13 and later
+        family = list(toolkit_font.families())
+    elif toolkit_font.family():
+        family.append(toolkit_font.family())
+    if toolkit_font.defaultFamily():
+        family.append(toolkit_font.defaultFamily())
```

Comment:
Is it okay to have duplicates in `family`? e.g. toolkit_font.family() returns the same thing as `toolkit_font.defaultFamily()`. This doesn't caught by FontFamily because it's a `CList`, not a `CSet`.

Reply:
Order matters here, so we don't want a `CSet`, and for Qt <= 5.12 we only take the first, so duplicates don't matter.

No idea what Qt does with duplicates in 5.13+ - I would hope that they are harmless.